addons/stock/models/stock_quant.py

```diff
@@ -759,19 +773,48 @@ def _get_available_quantity(self, product_id, location_id, lot_id=None, package_
             else:
                 return sum([available_quantity for available_quantity in availaible_quantities.values() if float_compare(available_quantity, 0, precision_rounding=rounding) > 0])
 
-    def _get_reserve_quantity(self, product_id, location_id, quantity, lot_id=None, package_id=None, owner_id=None, strict=False):
+    def _get_reserve_quantity(self, product_id, location_id, quantity, product_packaging_id=None, uom_id=None, lot_id=None, package_id=None, owner_id=None, strict=False, only_available=False):
```

Comment:
Do we really need the only_available? We always want to reserve all the available quantity no?

Reply:
I added it because before my changes, `_get_reserve_quantity` would reserve any quantity given without checking for the available qty.  If we remove `only_available` calling `_update_reserved_quantity` will be reserve the available quantity at most, which does make sense :thinking: